### DISCUSSION TITLE
Deployment in Heroku with PostGres : Error: no pg_hba.conf entry for host

### DISCUSSION THREAD

**Discussion by valarmathiarikrishnan** [strapi/strapi#5965] [Legacy Topics]
I'm trying to do a deployment of a new strapi project to heroku using PostGres. The deployment is successful but the application does not work.

When I connected postgres database locally, works fine without any issues. But when I connected with heroku postgres, I am getting the following error

ref: https://strapi.io/documentation/3.0.0-alpha.x/guides/deployment.html#heroku
```
yarn develop
yarn run v1.22.4
$ strapi develop
[2020-04-24T12:32:28.264Z] debug :no_entry:️ Server wasn't able to start properly.
[2020-04-24T12:32:28.265Z] error error: no pg_hba.conf entry for host ******** , user "********", database "********", SSL off
    at Connection.parseE (********/node_modules/pg/lib/connection.js:581:48)
    at Connection.parseMessage (********/node_modules/pg/lib/connection.js:380:19)
    at Socket.<anonymous> ********/node_modules/pg/lib/connection.js:116:22)
    at Socket.emit (events.js:311:20)
    at addChunk (_stream_readable.js:294:12)
    at readableAddChunk (_stream_readable.js:275:11)
    at Socket.Readable.push (_stream_readable.js:209:10)
    at TCP.onStreamRead (internal/stream_base_commons.js:186:23)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

```
**System**

Node.js version: v12.16.1
NPM version: 6.14.4
Strapi version: "3.0.0-beta.20",
Database: PostGres
Operating system: Ubuntu 18.04.3 LTS

Any idea on how to fix this thing?

**Comment by derrickmehaffy** [comment]
Based on a bit of research this is not a problem with Strapi, this issue relates to the database denying the connection as the connecting IP Address doesn't have access to the database.

https://dba.stackexchange.com/questions/193242/no-pg-hba-conf-entry-for-host-local-user-pgadmin-database-postgres-ss 

I'm moving this to a discussion as it's not a bug with Strapi.

**Comment by jmadelaine** [comment]
Strapi docs are wrong. Postgres requires all production databases to use SSL, and will deny unsecured connections. The Strapi docs here: https://strapi.io/documentation/v3.x/deployment/heroku.html say to set SSL to `false`, and set the node env to `production`. Doing this prevents Strapi connecting to postgres. Setting node env to `development` allows Strapi to connect to postgres.

**Reply by derrickmehaffy** [reply]
Was fixed in the move to the new docs repo: https://strapi.io/documentation/developer-docs/latest/deployment/heroku.html#_4-create-your-heroku-database-config-file-for-production

Locking thread as tested in the following repo on Heroku: https://github.com/derrickmehaffy/heroku-test-ee
